letsencrypt/models/letsencrypt.py

```diff
@@ -132,23 +135,45 @@ def generate_csr(self, domain):
             self.call_cmdline(cmdline)
         return csr
 
+    def _get_crt(self, csr):
+        """Get certificate for csr using the acme protocol."""
+        from acme_tiny import get_crt, DEFAULT_CA
+        account_key = self.generate_account_key()
+        acme_challenge = get_challenge_dir()
+        if not os.path.isdir(acme_challenge):
+            os.makedirs(acme_challenge)
+        return get_crt(
+            account_key,
+            csr,
+            acme_challenge,
+            log=_logger,
+            CA=DEFAULT_CA
+        )
+
     @api.model
     def cron(self):
+        save_error = None
         domain = urlparse.urlparse(
             self.env['ir.config_parameter'].get_param(
                 'web.base.url', 'localhost')).hostname
         self.validate_domain(domain)
-        account_key = self.generate_account_key()
-        csr = self.generate_csr(domain)
-        acme_challenge = get_challenge_dir()
-        if not os.path.isdir(acme_challenge):
-            os.makedirs(acme_challenge)
         if self.env.context.get('letsencrypt_dry_run'):
             crt_text = 'I\'m a test text'
         else:  # pragma: no cover
-            from acme_tiny import get_crt, DEFAULT_CA
-            crt_text = get_crt(
-                account_key, csr, acme_challenge, log=_logger, CA=DEFAULT_CA)
+            csr = self.generate_csr(domain)
+            try:
+                crt_text = self._get_crt(csr)
+            except Exception as ex:
+                # If certificate renewal fails for altname, we still will
+                # keep main domain with valid certificate:
+                _logger.warn(
+                    'Exception occured during certificate retrieval.'
+                    ' Retrying without altnames. Exception was: %s' %
+                    str(ex)
+                )
+                save_error = ex
+                csr = self.generate_csr(domain, ignore_altnames=True)
```

Comment:
No, because an exception raised in the exception handler, will not be covered by the previous try: So if there is a problem with web.base.url, the function will be ended by an exception immediately.